types/ckeditor__ckeditor5-engine/src/model/item.d.ts

```diff
@@ -1,4 +1,7 @@
 import Node from "./node";
 import TextProxy from "./textproxy";
+import Element from "./element";
+import RootElement from "./rootelement";
+import Text from "./text";
 
-export type Item = Node | TextProxy;
+export type Item = Node | Element | RootElement | Text | TextProxy;
```

Comment:
Same as before. `Element` and `Text` extend `Node`.

Reply:
It's true that they extend Node, but a union of subclasses and a super-type aren't the same when it comes to type narrowing. Consider this example:

```typescript
const foo = null as unknown as Node;
const bar = null as unknown as Node | Element;

if ("maxOffset" in foo) {
  const el: Element = foo; // TS2740: Type 'Node' is missing the following properties from type 'Element': childCount, isEmpty, maxOffset, name, and 7 more
}
if ("maxOffset" in bar) {
    const el: Element = bar; // No type error
}
```